File: bank-service/src/main/java/rs/edu/raf/repository/accounts/BankAccountRepository.java
package rs.edu.raf.repository.accounts;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.accounts.BankAccount;

import java.util.List;
import java.util.Optional;

@Repository
public interface BankAccountRepository extends JpaRepository<BankAccount,Long> {
    Optional<BankAccount> findByAccountNumber(Long accountNumber);
    List<BankAccount> findAllByOwner(Long owner);
    Optional<BankAccount> findByIdAndActiveIsTrue(Long id);
    Optional<BankAccount> findByAccountNumberAndActiveIsTrue(Long accountNumber);
    Optional<BankAccount> findByOwnerAndCurrency_Code(Long owner, String currencyCode);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/repository/accounts/BankAccountRepository.kt
package org.example.bankservicekotlin.repository.accounts

import org.example.bankservicekotlin.model.accounts.BankAccount
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface BankAccountRepository : JpaRepository<BankAccount, Long> {
    fun findByAccountNumber(accountNumber: Long): Optional<BankAccount>
    fun findAllByOwner(owner: Long): List<BankAccount>
    fun findByIdAndActiveIsTrue(id: Long): Optional<BankAccount>
    fun findByAccountNumberAndActiveIsTrue(accountNumber: Long): Optional<BankAccount>
    fun findByOwnerAndCurrency_Code(owner: Long, currencyCode: String): Optional<BankAccount>
    fun findAllByOwnerAndCurrency_IdAndActiveIsTrue(owner: Long, currencyId: Long): List<BankAccount>
}